File: 1532/Arremesso de Bolas.py
N, V = map(int, input().split())

while N!=0 and V!=0:
    possible = False

    for j in range(V, 0, -1):
        value = j
        i = j
        saltos = 1

        while value < N and i != 0:
            if saltos == i:
                saltos = 0
                i-=1
            
            value+=i
            saltos+=1

        
        if value == N:
            possible = True
            break
    

    if possible:
        print("possivel")
    else:
        print("impossivel")


    N, V = map(int, input().split())

File: 1532/Arremesso de Bolas.cpp
#include <bits/stdc++.h>
using namespace std;

int main (){

    int N, V;
    cin >> N >> V;
    
    while (N!=0 && V!=0){
        bool possible = false;
        for (int j = V; j > 0; j--){
            int value = j;
            int i = j;
            int saltos = 1;
            while (value < N && i!=0){



                if (saltos == i){
                    saltos = 0;
                    i--;
                }
                value = value + i;  
                saltos++;
                //cout << value << endl;
            }
            
            if (value == N){
                possible = true;
                break;
            }
        

        }
          
        if (possible) cout << "possivel\n";
        else cout << "impossivel\n";

        cin >> N >> V;
    }
}
